### DISCUSSION TITLE
Export from loaded Assetbundles doesn't include Materials/Textures

### DISCUSSION THREAD

**Discussion by StefanOber** [atteneder/glTFast#486] [Q&A]
Hello there,

we are loading Assetbundles in Unity 2022.2.0. When I try to export them to glb, the materials and textures are missing.

The validation report shows three error messages:

MESH_PRIMITIVE_ATTRIBUTES_ACCESSOR_INVALID_FORMAT - Invalid accessor format '{VEC4, UNSIGNED_BYTE}' for this attribute semantic. Must be one of ('{VEC3, FLOAT}', '{VEC3, UNSIGNED_BYTE normalized}', '{VEC3, UNSIGNED_SHORT normalized}', ..., '{VEC4, UNSIGNED_BYTE normalized}', '{VEC4, UNSIGNED_SHORT normalized}').

ACCESSOR_MIN_MISMATCH - Declared minimum value for this component (xxx) does not match actual minimum (xxx). | /accessors/0/min/2

ACCESSOR_ELEMENT_OUT_OF_MIN_BOUND - Accessor contains x element(s) less than declared minimum value xxx.
(second and third messge show up many times)

Are there any limitations with Assetbundles or is this another problem?

Thanks!

[Edit] We use HDRP in this project.

**Comment by atteneder** [comment]
@StefanOber Hi and thanks for reporting.

Which shaders to you use on those materials?

The validation errors seem to be mesh related. Would it be possible for you to provide a test prefab that reproduces the error?

Thanks

**Reply by StefanOber** [reply]
Hi atteneder,

thanks for the quick reply!

We use different shaders, some are custom (Shader Graph) but also the default HDRP/Lit shader. They all appear white.

Unfortunately I can't provide the prefab. But I just noticed, also a simple cube with a textured HDRP/Lit material appears white (no Assetbundle, just the editor cube). With no error message.

This is the code I used for exporting:

`var logger = new CollectingLogger();
            var exportSettings = new ExportSettings
            {
                format = GltfFormat.Binary,
                imageDestination = ImageDestination.MainBuffer,
                fileConflictResolution = FileConflictResolution.Overwrite
            };

            var export = new GameObjectExport(exportSettings, logger: logger);

            export.AddScene(new GameObject[] { Asset });

            bool success = await export.SaveToFileAndDispose(Application.streamingAssetsPath + "/gltf/test.glb");

            if (!success)
            {
                Debug.LogError("Something went wrong exporting a glTF");
            }`

**Reply by StefanOber** [reply]
I will import your package in an empty project and see if it works.

**Reply by atteneder** [reply]
For custom shaders you might need a custom implementation of `IMaterialExport`. It's not well documented though.

HDRP Lit should work though (just tested successfully). Can you provide a sample project or exported package with that cube of yours that does not work?

**Reply by StefanOber** [reply]
I just reinstalled the plugin and now it works with the default shaders. Sorry I don't know what went wrong at first. 
Just the custom shaders appear white. I will have a look at IMaterialExport.

Thanks for your help!

**Reply by atteneder** [reply]
Awesome!

At some point I'll look at custom shader workflows and document them properly.
